pjlib/src/pj/ssl_sock_ossl.c

```diff
@@ -1025,11 +1016,18 @@ static int verify_cb(int preverify_ok, X509_STORE_CTX *x509_ctx)
         break;
     }
 
-    /* When verification is not requested just return ok here, however
-     * application can still get the verification status.
+    /* When verification is not requested just return ok here.
+     * Otherwise, invoke app's verification callback, if configured.
+     * Anyway, application can still get the verification status.
      */
-    if (PJ_FALSE == ssock->param.verify_peer)
+    if (PJ_FALSE == ssock->param.verify_peer) {
         preverify_ok = 1;
+    } else if (ssock->param.cb.on_verify_cb) {
```

Comment:
Not sure if this is intentional, but there seems to be a behavior change with the patch.
Previously, the callback will be called regardless of the value of `verify_peer`, but now only if `verify_peer` is `PJ_TRUE`.

Reply:
Forgot to mention this in the desc, yes it is intentional. The callback invocation needs to be moved down to get preverification status. Then I thought the callback should also regard the `verify_peer` setting as otherwise there may be contradiction, i.e: TLS connection may be dropped due to verification while `verify_peer` is disabled (when callback implementation in app does not check if verification is disabled).

Alternatively, we can add some docs in `pj_ssl_sock_param.verify_peer` and `pjsip_tls_setting.verify_client/server` to explain the behavior, i.e: if verification callback is set/implemented, those flags will be ignored.